src/storage.rs

```diff
@@ -136,7 +144,7 @@ impl From<Bytecode> for EvmCode {
                 delegated_address: code.delegated_address,
                 version: code.version,
             }),
-            Bytecode::Eof(_) => unimplemented!("TODO: Support EOF"),
+            Bytecode::Eof(code) => EvmCode::Eof(EofCode(code.raw.clone())),
```

Comment:
We don't need to clone here, can write this as `Arc::unwrap_or_clone(code).raw`.